gc/src/trace.rs

```diff
@@ -105,10 +107,60 @@ macro_rules! custom_trace {
     };
 }
 
-impl<T: ?Sized> Finalize for &'static T {}
-unsafe impl<T: ?Sized> Trace for &'static T {
+/// A marker trait for types that don't require tracing.
+/// 
+/// # Safety
+/// TODO: Safety conditions
+pub unsafe trait EmptyTrace {}
+
+// TODO: The README needs to be updated to explain when `Rc` and the other types here can be managed by GC.
+impl<T: EmptyTrace + ?Sized> Finalize for Rc<T> {}
```

Comment:
```suggestion
impl<T: ?Sized> Finalize for Rc<T> {}
```

since `Finalize` itself doesn’t recursively finalize inner members (we have `Trace::finalize_glue` for that).

Same for the rest.